helm-charts/splunk-otel-collector/templates/operator/instrumentation.yaml

```diff
@@ -19,29 +19,30 @@ spec:
     - tracecontext
     - baggage
     - b3
-  {{- if .Values.operator.instrumentation.spec.sampler }}
-  {{ include .Values.operator.instrumentation.spec.sampler }}
+  {{- with .Values.operator.instrumentation.spec.sampler }}
+  sampler:
+  {{- toYaml . | nindent 4 }}
   {{- end }}
   env:
-    {{- if .Values.operator.instrumentation.spec.env }}
-    {{- include .Values.operator.instrumentation.spec.env }}
-    {{- end }}
-    {{- if .Values.splunkObservability.profilingEnabled }}
-    {{- if not hasKey (include "splunk-otel-collector.operator.extract-name-keys-from-dict-list" .Values.operator.instrumentation.spec.env) "SPLUNK_PROFILER_ENABLED" }}
+  {{- with .Values.operator.instrumentation.spec.env }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- if .Values.splunkObservability.profilingEnabled }}
+  {{- if eq (include "splunk-otel-collector.operator.env-has" (dict "env" .Values.operator.instrumentation.spec.env "envName" "SPLUNK_PROFILER_ENABLED")) "false" }}
     - name: SPLUNK_PROFILER_ENABLED
       value: "true"
-    {{- end }}
-    {{- if not hasKey (include "splunk-otel-collector.operator.extract-name-keys-from-dict-list" .Values.operator.instrumentation.spec.env) "SPLUNK_PROFILER_MEMORY_ENABLED" }}
+  {{- end }}
+  {{- if eq (include "splunk-otel-collector.operator.env-has" (dict "env" .Values.operator.instrumentation.spec.env "envName" "SPLUNK_PROFILER_MEMORY_ENABLED")) "false" }}
```

Comment:
Can we just set the following as the default in the values.yaml instead of applying this magic under the hood?
```yaml
operator:
  instrumentation:
    spec:
      env:
        - name: SPLUNK_PROFILER_ENABLED
          value: "true"
        - name: SPLUNK_PROFILER_MEMORY_ENABLED
          value: "true"
```

Reply:
Unfortunately I would not recommend this without other changes or trade offs.
- If we instrument applications with profiling enabled and the collector isn't setup to receiver profiling data (splunkObservability.profilingEnabled: true) , the instrumented application will waste resources and log error messages.
- I think it is easier for customers to get lost if we don't apply some magic under the hood.
  - We tried using only validation rules at the start, this was not trivial so we moved onto our current implementation.
  - I wish we could use [yaml-anchors](https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors) in the values file and link the value of splunkObservability.profilingEnabled to these env vars. However since profilingEnabled would be a boolean and these env vars have to have string values it is either not possible or would require some magic.

Reply:
Unfortunately I would not recommend this without other changes or trade offs.
- If we instrument applications with profiling enabled and the collector isn't setup to receiver profiling data (splunkObservability.profilingEnabled: true) , the instrumented application will waste resources and log error messages.
- I think it is easier for customers to get lost if we don't apply some magic under the hood.
  - We tried using only validation rules at the start, this was not trivial so we moved onto our current implementation.
  - I wish we could use [yaml-anchors](https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors) in the values file and link the value of splunkObservability.profilingEnabled to these env vars. However since profilingEnabled would be a boolean and these env vars have to have string values it is either not possible or would require some magic.

Reply:
Makes sense. Thanks for clarifying